qiskit/circuit/parameterexpression.py

```diff
@@ -266,3 +311,12 @@ def __eq__(self, other):
         return (isinstance(other, ParameterExpression)
                 and self.parameters == other.parameters
                 and srepr(self._symbol_expr) == srepr(other._symbol_expr))
+
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        if '_lambda' in state:
```

Comment:
In addition to the pickling issue, it looks like this will also address the potential issue around `exec`-ing untrusted input with `lambdify` ( from https://docs.sympy.org/latest/modules/utilities/lambdify.html ). Can you check that this is the case, and add a comment explaining that we need to exclude `_lambda` both for pickling and for security?

Reply:
@kdk do you know ways to prevent unsanitized input for `lambdify`?
Can we say `self._symbol_expr` is sanitized if `self._symbol_expr.subs` works without errors?

Reply:
My concern on lambdify is that it has the potential for executing raw string input as code. My concern from a security PoV would be someone constructing a parameter with something like `import os; os.removedirs('/')` or some other arbitrary command and when subs is called it runs. But, I quickly tested this locally and couldn't get it to execute an aribtrary command (although I might not understand the full path to call this). But it looks like all the input being wrapped in a `sympy.Symbol` is preventing strings from being evaluated as python code.

Reply:
@mtreinish Thank you for your comments. Currently, all the inputs must be `numbers.Real` because of call of `self._raise_if_passed_non_real_value(parameter_values)`. So, there is no way to pass raw string to `self._lambda`, I believe.

Reply:
I'm not sure if calling `self._symbol_expr.subs` once will be sufficient to say that an expression is sanitized. Within python, the most likely concern is a parameter name escaping the lambda definition during the exec, so I'm not sure how calling `subs` guards against this.